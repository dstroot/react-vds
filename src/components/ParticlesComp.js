import React, { Component } from 'react';
import Responsive from 'react-responsive-decorator'
import Particles from 'particlesjs'
  
class ParticlesComp extends Component {
  
  constructor (props) {
    super(props)
    this.state = {
      isHidden: false
    }
  }
  
  componentDidMount() {
    this.props.media({ minWidth: 768 }, () => {
      this.setState({
        isHidden: false
      });
      
      Particles.init({
        selector: '.background',
        maxParticles: 70,
        connectParticles: true,
        color: '#7952b3',
      });
    });
    
    this.props.media({ maxWidth: 768 }, () => {
      this.setState({
        isHidden: true
      });
    });
  }

  componentWillUnmount() {
    this.props.media({ minWidth: 768 }, () => {
      Particles.destroy();
    });
  }

  render() {
    const {isHidden} = this.state;
    console.log(isHidden);
    return (
      <div>
        {isHidden ? <div></div> :
        <div className="canvas-container">
          <canvas className="background" id="canvas"></canvas>
        </div>}
      </div>
    );
  }
}

// we can use the responsive decorator as a HOC
export default Responsive(ParticlesComp);