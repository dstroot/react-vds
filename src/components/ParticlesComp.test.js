import React from 'react';
import ParticlesComp from './ParticlesComp';
import renderer from 'react-test-renderer';

test('ParticlesComp renders the terms page', () => {
  const component = renderer.create(<ParticlesComp />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
