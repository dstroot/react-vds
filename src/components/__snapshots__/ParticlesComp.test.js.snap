// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ParticlesComp renders the terms page 1`] = `
<div>
  <div
    className="canvas-container"
  >
    <canvas
      className="background"
      id="canvas"
    />
  </div>
</div>
`;
